File: learning-javascript/leet-code/src/best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.ts
export function maxProfit(prices: number[]): number {
	let max = 0;
	let dayBought = 0;
	let daySold = 1;
	 while(daySold < prices.length) {
		if(prices[dayBought] < prices[daySold]){
			const profit = prices[daySold] - prices[dayBought];
			max = Math.max(profit, max);		
		} else {
			dayBought = daySold;
		}
		daySold++;
	}
 return max;
}

export function maxProfit2(prices: number[]): number {
	let profits = 0; 
	let dayBought = 0;
	let daySold = 1;
	 while(daySold < prices.length) {
		if(prices[dayBought] < prices[daySold]){
			const profit = prices[daySold] - prices[dayBought];
			profits += profit;
		}
		dayBought = daySold;
		daySold++;
	}
 return profits;
}

File: katas/bestTimeToBuyAndSellStock/bestTimeToBuyAndSellStock.py
def maxProfit(prices):
	max_profit = 0
	dayBought = 0
	daySold = 1
	while daySold < len(prices): 
		if prices[dayBought] < prices[daySold]: 
			profit = prices[daySold] - prices[dayBought]
			max_profit = max(profit, max_profit)
		else:
			dayBought = daySold
		daySold += 1
	return max_profit

def maxProfit2(prices):
	max_profit = 0
	dayBought = 0
	daySold = 1
	while daySold < len(prices): 
		if prices[dayBought] < prices[daySold]: 
			profit = prices[daySold] - prices[dayBought]
			max_profit += profit
		dayBought = daySold
		daySold += 1
	return max_profit